src/components/Modal/Modal.js

```diff
@@ -1,111 +1,113 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
+import FocusTrap from 'focus-trap-react';
+import ModalHeader from './ModalHeader';
+import { getContentId, getTitleId } from './utils';
 
-const Modal = (props) => {
-  const {
-    children,
-    className,
-    descriptionId,
-    dialog,
-    label,
-    large,
-    open,
-    prompt,
-    ...rest
-  } = props;
+class Modal extends Component {
+  onClose = (evt) => {
+    evt.stopPropagation();
+    const { onClose } = this.props;
+    onClose();
+  }
 
-  const isOpen = !!open;
-  const isDialog = !!dialog || !!prompt;
-  const role = prompt ? 'alertdialog' : 'dialog';
-  const containerRole = isDialog ? 'document' : null;
+  onKeyUp = (evt) => {
+    const { key } = evt;
+    if (key === 'Escape') { this.onClose(evt); }
+  }
 
-  const childrenWithProps = [...children].map((child) => {
-    const childName = child ? child.type.displayName || child.type.name : null;
+  cloneWithProps = (child) => {
+    const name = child.type.displayName || child.type.name;
 
-    if (childName === 'ModalHeader') {
-      return React.cloneElement(child, {
-        key: label,
-        label,
-        prompt,
-        uncloseable: child.props.uncloseable !== undefined ? child.props.uncloseable : !!prompt,
-      });
+    if (name === 'ModalContent') {
```

Comment:
Is something like `if (child.type === ModalHeader) { ... }` possible (without the hardcoded component name)?

Reply:
Sorta, but checking for a string is more performant here

Reply:
Why?

Reply:
Because a React Component is an object and shallow comparing objects of unknown shape is more expensive than comparing two strings.

Reply:
Alright, I thought that `ModalHeader` would be a function and that `child.type` would be exactly that function, but you're probably right ✅